GenerateOtp/redis.ts

```diff
@@ -0,0 +1,83 @@
+import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
+import { FiscalCode } from "@pagopa/ts-commons/lib/strings";
+import { toError, tryCatch2v } from "fp-ts/lib/Either";
+import { none, Option, some } from "fp-ts/lib/Option";
+import {
+  fromEither,
+  fromPredicate,
+  TaskEither,
+  taskEither
+} from "fp-ts/lib/TaskEither";
+import * as t from "io-ts";
+import { RedisClient } from "redis";
+import { Otp } from "../generated/definitions/Otp";
+import { OtpCode } from "../generated/definitions/OtpCode";
+import { Timestamp } from "../generated/definitions/Timestamp";
+import { errorsToError } from "../utils/conversions";
+import { getTask, setWithExpirationTask } from "../utils/redis_storage";
+
+export const OtpPayload = t.interface({
+  expiresAt: Timestamp,
+  fiscalCode: FiscalCode,
+  ttl: NonNegativeInteger
+});
+
+export type OtpPayload = t.TypeOf<typeof OtpPayload>;
+
+const OTP_FISCAL_CODE_PREFIX = "OTP_FISCALCODE_";
+const OTP_PREFIX = "OTP_";
+
+export const storeOtpAndRelatedFiscalCode = (
+  redisClient: RedisClient,
+  otpCode: OtpCode,
+  payload: OtpPayload,
+  otpTtl: NonNegativeInteger
+): TaskEither<Error, boolean> =>
+  setWithExpirationTask(
+    redisClient,
+    `${OTP_PREFIX}${otpCode}`,
+    JSON.stringify(payload),
+    otpTtl
+  )
+    .chain(
+      fromPredicate(
+        _ => _,
+        () => new Error("Cannot Store OTP")
+      )
+    )
```

Comment:
We should use more strict typings here. 
```suggestion
): TaskEither<Error, true> =>
  setWithExpirationTask(
    redisClient,
    `${OTP_PREFIX}${otpCode}`,
    JSON.stringify(payload),
    otpTtl
  )
    .chain(
      fromPredicate(
        (_: boolean): _ is true => _,
        () => new Error("Cannot Store OTP")
      )
    )
```
We need to change even the return type of `setWithExpirationTask` as `TaskEither<Error, true>`

Reply:
Sorry. I see now that if we change the return type of `setWithExpirationTask` we can remove the `chain` with the `fromPredicate`

Reply:
Done in [94ca09e](https://github.com/pagopa/io-functions-cgn/pull/35/commits/94ca09e6e229acfe3c8dbd59cbe76c0e52c21b7b). Note that `_ => _` is correcty inferred to `true`

Reply:
>Sorry. I see now that if we change the return type of setWithExpirationTask we can remove the chain with the fromPredicate. YEP You're right :)